src/main/java/duke/task/Event.java

```diff
@@ -0,0 +1,125 @@
+                        + " to " + atTimeEnd.format(DateTimeFormatter.ofPattern("HH:mm")) + ")";
+            }
+        }
+    }
+
+    /**
+     * Stringifies the Event object to a special string representation which includes the number of days to
+     * the event (if event has not passed), or the number of days since the event (if event has passed).
+     *
+     * @return Formatted string representation of the Event object
+     */
+    public String displayEventTime() {
+        try {
+            if (atTimeStart == null) {
+                return "[E]" + super.toString() + " (at: "
+                        + atDate.format(DateTimeFormatter.ofPattern("d MMM yyyy")) + ") - "
+                        + LocalDate.now().datesUntil(atDate).count() + " day(s) to go!";
+            } else {
+                if (atTimeEnd == null) {
+                    return "[E]" + super.toString() + " (at: "
+                            + atDate.format(DateTimeFormatter.ofPattern("d MMM yyyy"))
+                            + ", " + atTimeStart.format(DateTimeFormatter.ofPattern("HH:mm")) + ") - "
+                            + LocalDate.now().datesUntil(atDate).count() + " day(s) to go!";
+                } else {
+                    return "[E]" + super.toString() + " (at: "
+                            + atDate.format(DateTimeFormatter.ofPattern("d MMM yyyy"))
+                            + ", " + atTimeStart.format(DateTimeFormatter.ofPattern("HH:mm"))
+                            + " to " + atTimeEnd.format(DateTimeFormatter.ofPattern("HH:mm")) + ") - "
+                            + LocalDate.now().datesUntil(atDate).count() + " day(s) to go!";
+                }
+            }
+        } catch (IllegalArgumentException iae) {
+            if (atTimeStart == null) {
+                return "[E]" + super.toString() + " (at: "
+                        + atDate.format(DateTimeFormatter.ofPattern("d MMM yyyy")) + ") - "
+                        + "it's " + atDate.datesUntil(LocalDate.now()).count() + " day(s) past the event!";
+            } else {
+                if (atTimeEnd == null) {
+                    return "[E]" + super.toString() + " (at: "
+                            + atDate.format(DateTimeFormatter.ofPattern("d MMM yyyy"))
+                            + ", " + atTimeStart.format(DateTimeFormatter.ofPattern("HH:mm")) + ") - "
+                            + "it's " + atDate.datesUntil(LocalDate.now()).count() + " day(s) past the event!";
+                } else {
+                    return "[E]" + super.toString() + " (at: "
+                            + atDate.format(DateTimeFormatter.ofPattern("d MMM yyyy"))
+                            + ", " + atTimeStart.format(DateTimeFormatter.ofPattern("HH:mm"))
+                            + " to " + atTimeEnd.format(DateTimeFormatter.ofPattern("HH:mm")) + ") - "
+                            + "it's " + atDate.datesUntil(LocalDate.now()).count() + " day(s) past the event!";
+                }
```

Comment:
You can consider implementing a Parser for DateTime formatting specifically to declutter your code of the numerous DateTimeFormatter lines, but it is up to your personal preference! 👍